int32 agentId
int32 action
